From 924cd5bb044f212900ac70131e8757795cebbfc5 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 6 Nov 2020 10:11:43 +0800
Subject: [PATCH] drm/rockchip: vop2: Fix some compile warning

Fix warning:

drivers/gpu/drm/rockchip/rockchip_drm_vop2.c:542: warning: Function
parameter or member 'phys_id' not described in
'vop2_find_win_by_phys_id'
drivers/gpu/drm/rockchip/rockchip_drm_vop2.c:542: warning: Function
parameter or member 'vop2' not described in 'vop2_find_win_by_phys_id'

drivers/gpu/drm/rockchip/rockchip_drm_vop2.c:3503
vop2_plane_create_name_property() warn: should '1 << (win->win_id)' be a
64 bit type?
drivers/gpu/drm/rockchip/rockchip_drm_vop2.c:3509
vop2_plane_create_name_property() warn: should '1 << (win->win_id)' be a
64 bit type?

Fixes: efe0578ec8a8 ("drm/rockchip: Add support for vop2")
Change-Id: Ia52eac3fcf2ddeff825b5fdaa26adcf26c71a7af
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 3676eea603ca..a5bdbe4ff768 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -533,7 +533,7 @@ static inline uint32_t vop2_get_intr_type(struct vop2 *vop2, const struct vop_in
 	return ret;
 }
 
-/**
+/*
  * phys_id is used to identify a main window(Cluster Win/Smart Win, not
  * include the sub win of a cluster or the multi area) that can do
  * overlay in main overlay stage.
@@ -3496,17 +3496,18 @@ static int vop2_plane_create_name_property(struct vop2 *vop2, struct vop2_win *w
 {
 	struct drm_prop_enum_list *props = vop2->plane_name_list;
 	struct drm_property *prop;
+	uint64_t bits = BIT(win->win_id);
 
 	prop = drm_property_create_bitmask(vop2->drm_dev,
 					   DRM_MODE_PROP_IMMUTABLE, "NAME",
 					   props, vop2->registered_num_wins,
-					   BIT(win->win_id));
+					   bits);
 	if (!prop) {
 		DRM_DEV_ERROR(vop2->dev, "create Name prop for %s failed\n", win->name);
 		return -ENOMEM;
 	}
 	win->name_prop = prop;
-	drm_object_attach_property(&win->base.base, win->name_prop, BIT(win->win_id));
+	drm_object_attach_property(&win->base.base, win->name_prop, bits);
 
 	return 0;
 }
@@ -3693,7 +3694,7 @@ static int vop2_create_crtc(struct vop2 *vop2)
 		drm_object_attach_property(&crtc->base, vop2->vp_id_prop, vp->id);
 	}
 
-	/**
+	/*
 	 * change the unused primary window to overlay window
 	 */
 	for (; j < vop2->registered_num_wins; j++) {
-- 
2.35.3

